orca-clouddriver/src/test/java/com/netflix/spinnaker/orca/clouddriver/config/CloudDriverConfigurationTest.java

```diff
@@ -0,0 +1,108 @@
+import java.util.List;
+import okhttp3.OkHttpClient;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ContextConfiguration;
+import retrofit.RequestInterceptor;
+import retrofit.RestAdapter;
+
+@ExtendWith(MockitoExtension.class)
+@ContextConfiguration(
+    classes = CloudDriverConfigurationProperties.class,
+    initializers = CloudDriverConfigurationTest.class)
+@EnableConfigurationProperties
+@SpringBootTest
+public class CloudDriverConfigurationTest extends YamlFileApplicationContextInitializer {
+
+  @Autowired private CloudDriverConfigurationProperties cloudDriverConfigurationProperties;
+
+  private CloudDriverConfiguration.ClouddriverRetrofitBuilder clouddriverRetrofitBuilder;
+
+  @Override
+  protected String getResourceLocation() {
+    return "classpath:clouddriver-sharding-properties.yml";
+  }
+
+  @BeforeEach
+  public void setup() {
+    OkHttpClientBuilderProvider okHttpClientBuilderProvider =
+        new OkHttpClientBuilderProvider() {
+          @Override
+          @NotNull
+          public Boolean supports(@NotNull ServiceEndpoint service) {
+            return true;
+          }
+
+          @Override
+          @NotNull
+          public OkHttpClient.Builder get(@NotNull ServiceEndpoint service) {
+            return new OkHttpClient().newBuilder();
+          }
+        };
+    OkHttpClientProvider okHttpClientProvider =
+        new OkHttpClientProvider(List.of(okHttpClientBuilderProvider));
+
+    ObjectMapper objectMapper = new ObjectMapper();
+    RestAdapter.LogLevel logLevel = RestAdapter.LogLevel.FULL;
+    RequestInterceptor requestInterceptor = new SpinnakerRequestInterceptor(null);
+
+    this.clouddriverRetrofitBuilder =
+        new CloudDriverConfiguration.ClouddriverRetrofitBuilder(
+            objectMapper,
+            okHttpClientProvider,
+            logLevel,
+            requestInterceptor,
+            cloudDriverConfigurationProperties);
+  }
+
+  @DisplayName("when selector config is a YAML list, successfully constructs a SelectableService")
+  @Test
+  public void testConstructingSelectorsWithYAMLList() {
+    /*
+     * Because the configuration of a BaseUrl in Orca is loosely typed as a Map<String, Object>, to allow variations of
+     * config that require different types, Jackson doesn't have enough type information and converts a YAML list to a
+     * LinkedHashMap. This caused issues in Kork where the ServiceSelector constructors expected a Map<String, Object>
```

Comment:
I think we're getting there....maybe the first paragraph of this comment could live the test yml file since that's where the expectation for a list comes from?

After staring at this in the debugger, I also think it would help to add some detail about what the LinkedHashMap built from the yaml list looks like.  But then...still trying to clarify that we're not talking about what orca passes as the `config` constructor parameter -- it's a Map<String, Object>.  We're talking about the type of the individual values.  So far I've got:

```suggestion
     * LinkedHashMap (where the keys are 0, 1, 2, 3, etc. and the values are the list elements). This test verifies that the ServiceSelector constructors treat the values of `Map<String, Object> config` properly -- as Maps.
```
an then maybe the next two lines can disappear?

And yes, the test is failing (as expected until https://github.com/spinnaker/kork/pull/1012), so that's good.

Reply:
Just looked at it again, and you're right. It's not the Map<String, Object> that's the issue, it's the Object within the map being a Map rather than a List.

I'll clarify that tomorrow.

Reply:
I think the comment does a better job explaining now.

Reply:
Agreed, thank you!